apps/crud/src/app/app.store.spec.ts

```diff
@@ -0,0 +1,145 @@
+import { TestBed } from '@angular/core/testing';
+import { render } from '@testing-library/angular';
+
+import { AppStore } from './app.store';
+import { TodoService } from './service/todo.service';
+import { createMockWithValues } from '@testing-library/angular/jest-utils';
+import { AppComponent } from './app.component';
+import { provideComponentStore } from '@ngrx/component-store';
+import { Todo } from './interfaces/Todo';
+import { lastValueFrom, take, throwError } from 'rxjs';
+
+describe('AppStore', () => {
+  it('truthy', async () => {
+    const { store } = await setup();
+
+    store.patchState({ todos: getFakeTodos(), callState: 'LOADED' });
+
+    expect(store).toBeTruthy();
+    const callState = await lastValueFrom(store.callState$.pipe(take(1)));
+    expect(callState).toEqual('LOADED');
+  });
+
+  describe('todos$ selector', () => {
+    it('should return todos from state', async () => {
+      const todos: Todo[] = getFakeTodos();
+      const { store } = await setup();
+
+      store.patchState({ todos: getFakeTodos(), callState: 'LOADED' });
+
+      const t = await lastValueFrom(store.todos$.pipe(take(1)));
+
+      expect(t.length).toEqual(todos.length);
+    });
+  });
+
+  // can't do success and failure
+  // setup function has only one mocked returned value
+  // and you can't overwrite it inside a test
```

Comment:
you can pass argument to your setup function. That's why it's better than a beforeEch